sync/syncer.go

```diff
@@ -329,33 +345,40 @@ func (s *Syncer) run() {
 func (s *Syncer) synchronise() {
 	// only one concurrent synchronise
 	if s.syncLock.TryLock() == false {
+		s.Log.Info("cannot sync because another sync is in progress")
 		return
 	}
-
 	// release synchronise lock
 	defer s.syncLock.Unlock()
-	curr := s.GetCurrentLayer()
 
+	curr := s.GetCurrentLayer()
 	// node is synced and blocks from current layer have already been validated
 	if curr == s.ProcessedLayer() {
-		s.Debug("node is synced")
+		s.Info("node is fully synced")
 		// fully-synced, make sure we listen to p2p
 		s.setGossipBufferingStatus(done)
 		return
 	}
 
 	// we have all the data of the prev layers so we can simply validate
 	if s.weaklySynced(curr) {
-		s.handleWeaklySynced()
-		if err := s.syncEpochActivations(curr.GetEpoch()); err != nil {
-			if curr.GetEpoch().IsGenesis() {
-				s.With().Info("cannot fetch epoch atxs (expected during genesis)", curr, log.Err(err))
-			} else {
-				s.With().Error("cannot fetch epoch atxs", curr, log.Err(err))
+		if len(s.net.GetPeers()) != 0 {
+			s.Log.Info("start syncing epoch atxs")
+			err := s.fetcher.GetEpochATXs(curr.GetEpoch()) //syncEpochActivations(curr.GetEpoch())
+			if err != nil {
+				if curr.GetEpoch().IsGenesis() {
```

Comment:
should we use `GetEffectiveGenesis` here instead?

Reply:
no, in genesis epochs this shouldn't be an error - in non genesis atxs are generated and therefore this should be an error